src/Microsoft.AspNetCore.SystemWebAdapters/Adapters/ISystemWebAdapterRemoteAppBuilder.Shared.cs

```diff
@@ -0,0 +1,11 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.AspNetCore.SystemWebAdapters;
+
+public interface ISystemWebAdapterRemoteAppBuilder
```

Comment:
What if we had two remote app builder interfaces - a client version and a server version. They could have the same shape (in fact, they could both just derive from `ISystemWebAdapterRemoteAppBuilder` with no changes) but then all of the methods that `ISystemWebAdapterRemoteAppBuilder` wouldn't need client or server in their names. You could just have `AddRemoteAppAuthentication` and have the client one extend `ISystemWebAdapterRemoteAppClientBuilder` while the server one extends `ISystemWebAdapterRemoteAppServerBuilder`.

Reply:
those type names don't show up anywhere, though, so I'm not sure how helpful that would be. I was wondering if we even needed the "client/server" after the initial one, i.e.

**ASP.NET Core:**

```csharp
builder.Services.AddSystemWebAdapters()
  .AddRemoteAppClient(remote => remote
    .Configure(options =>
    {
      options.RemoteAppUrl = ...
      options.ApiKey = ...
    }
    .AddAuthentication(true)
    .AddSession());
```

**ASP.NET Framework:**

```csharp
SystemWebAdapterConfiguration.AddSystemWebAdapters(this)
    .AddRemoteAppServer(remote => remote
        .Configure(options => options.ApiKey = ... )
        .AddAuthentication()
        .AddSession());
```

It's already being done as part of `AddRemoteApp[Server|Client]` so this seems like it would be fairly clear.

Reply:
Even if the user doesn't see the type names, I think it would help because it would allow us to have the API you're describing but only the correct `Configure` or `AddAuthentication` would be usable (and suggested by Intellisense) because they would extend different types. As it is now, wouldn't the user have to choose between to `Configure` methods? That's not a great experience which is why we were going to name them Client and Server. I think if they extend different types, though, then we can get away with them both just being called 'Configure'

Reply:
Oh - the client stuff is only available on the core side while the framework is only available on the framework. But I see your point. I'll try that out

Reply:
Ah, good point. I forgot that even though `ISystemWebAdapterRemoteApp` is shared, the extension methods aren't! In that case, it doesn't make as big of a difference. I still think it would be nice to have client and server versions of the builder so that it's clearer which the extension methods apply to without having to give them cumbersome names.

Reply:
I made a separate builder for both